### DISCUSSION TITLE
Search for specimens by month?

### DISCUSSION THREAD

**Discussion by AFisherCSULB** [Symbiota/community-board#45] [Wish List]
I'm wondering if there are operators I can use in the Collection Date search text box to find specimens collected in particular months? 

For example, let's say I want to find all of the specimens collected from a park in Los Angeles County across all years in the months of September and October.  Is there already a way to search for this using a wild card for the year? 

Thank you! 
Amanda

**Comment by themerekat** [comment]
I've wanted this as well! I don't think it is currently possible.

**Comment by mickley** [comment]
@AFisherCSULB:

You actually can already do this, just by a more roundabout way. In the occurrence editor search, find Custom Field 1, and select Date in the dropdown of fields. Then, select CONTAINS, and in the box put `-09-`. That will return all of the records from September. 

To also include October, click the pencil at the end of Custom Field 1 to add a second custom field, set it to OR, and set up as before but with `-10-`.

**Comment by mickley** [comment]
@themerekat: This would not be difficult to add to the main date field either

MySQL supports queries like so: `WHERE eventdate LIKE '%-06-%'`, which is effectively what is happening behind the scenes for the custom field solution above. 

All that would need to be done is the following:
1). Add another elseif statement to detect percent wildcards and use `LIKE` instead of `=` in OccurrenceEditorManager.php here: https://github.com/BioKIC/Symbiota/blob/09c025ccae1ad5989f07ed3d727d0f2ad8c1755d/classes/OccurrenceEditorManager.php#L404
2). Add another validformat to collections.editor.query.js to allow dates that include wildcards, see here: https://github.com/BioKIC/Symbiota/blob/master/js/symb/collections.editor.query.js

I can do this at some point if you are interested.

**Comment by themerekat** [comment]
Are LIKE searches slower than = ? That would be my only hesitation. Otherwise, sounds fantastic! It would be great to have that capability.

**Reply by mickley** [reply]
Probably, but the difference may be slight. It would depend on the indexing setup on Symbiota though. 

See, for example: https://stackoverflow.com/questions/6142235/sql-like-vs-performance

In practice, on our portal, it's maybe a second to do a search by month(s). So, not noticeably slow.
